### PR TITLE
OnlineStore-2/OnlineStore-10 (added models)

### PR DESCRIPTION
**Pull request by SergeyBrykulskii**
https://app.clickup.com/t/8694v8z75
https://app.clickup.com/t/8694v90wq

### CODE REVIEW

File: OnlineStore/OnlineStore.Domain/Result/BaseResult.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace OnlineStore.Domain.Result;
+
+public class BaseResult<T>
+{
+    public bool IsSuccess => ErrorMessage == null;
+    public string? ErrorMessage { get; set; }
+    public int? ErrorCode { get; set; }
+    public T? Data { get; set; }
+}
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Domain/Result/BaseResult.cs]: why is it on the domain layer

File: OnlineStore/OnlineStore.Domain/Entities/Order.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using OnlineStore.Domain.Interfaces.Entities;
+
+namespace OnlineStore.Domain.Entities;
+
+public class Order : IEntity
+{
+    public long Id { get; set; }
+    public DateTime CreatedAt { get; set; }
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Domain/Entities/Order.cs]: Datetime.Now

File: OnlineStore/OnlineStore.Domain/Entities/OrderProduct.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace OnlineStore.Domain.Entities;
+
+public class OrderProduct
+{
+    public long ProductId { get; set; }
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Domain/Entities/OrderProduct.cs:8]: missing Quantity

File: OnlineStore/OnlineStore.Domain/Entities/Category.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using OnlineStore.Domain.Interfaces.Entities;
+
+namespace OnlineStore.Domain.Entities;
+
+public class Category : IEntity
+{
+    public long Id { get; set; }
+    public string Name { get; set; }
+    public List<Product>? Products { get; set; }
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Domain/Entities/Category.cs:9]: Why nullable

File: OnlineStore/OnlineStore.Domain/Entities/Order.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    public List<Product> Products { get; set; }
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Domain/Entities/Order.cs]: OrderProduct

File: OnlineStore/OnlineStore.Domain/Entities/Product.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+public class Product : IEntity
+{
+    public long Id { get; set; }
+    public string Name { get; set; }
+    public string Description { get; set; }
+    public double Price { get; set; }
+    public long CategoryId { get; set; }
+    public Category Category { get; set; }
+    public List<Order>? Orders { get; set; }
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Domain/Entities/Product.cs]: remove this property

File: OnlineStore/OnlineStore.Domain/Entities/User.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿using OnlineStore.Domain.Interfaces.Entities;
+
+namespace OnlineStore.Domain.Entities;
+
+public class User : IEntity
+{
+    public long Id { get; set; }
+    public string Name { get; set; }
+    public string Email { get; set; }
+    public string Password { get; set; }
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Domain/Entities/User.cs]: passwordHash

**Review by SUPERKISA8BIIT — COMMENTED** [review]
checked

File: OnlineStore/OnlineStore.Domain/Result/BaseResult.cs (hunk 1 above)
Comment (SergeyBrykulskii) [reply OnlineStore/OnlineStore.Domain/Result/BaseResult.cs]: where it should be? application?

**Pull request merged**